recipe/meta.yaml

```diff
@@ -30,15 +30,16 @@ requirements:
     - pip
     - wheel
     - numpy
-    - cython
+    - cython =3.0.2
     - extension-helpers
   run:
     - python
     - astropy
-    - {{ pin_compatible('numpy') }}
+    - numpy <2
```

Comment:
Shouldn't this PR then closed as the package is not np2 compatible?

Reply:
I gather from the comments the bot made when the PR was opened that in the event of failure to build this PR should be left open until there is a numpy 2 compatible version.

It would probably make sense to do a new build that includes the numpy upper bound in a separate PR, though...

Reply:
Hey thanks for helping with this update. The latest release is definitely not numpy 2 compatible. There is some problem with integer dtypes in the cython code:

```
cdef vector[np.int_t] j_ind
unknown type in template argument
```
There is a short comment that appears closely related to this in the [Numpy 2 migration docs](https://numpy.org/devdocs/numpy_2_0_migration_guide.html), but I haven't figured out how to resolve it yet.

Reply:
I think @bsipocz is correct that this PR should be closed since halotools is not compatible with numpy 2.0 yet. I've just now released [v0.9.1 on pypi](https://pypi.org/project/halotools/0.9.1/) - this does not resolve the numpy 2.0 incompatibility, but it finally updates the packaging to be fully based on `pyproject.toml`, so hopefully that makes life easier going forward.